lib/MooseX/Role/FFI.pm

```diff
@@ -0,0 +1,69 @@
+package MooseX::Role::FFI;
+# ABSTRACT: Easily create interfaces to FFI functions with Moose roles
+
+use Moose::Role;
+use FFI::Platypus 0.89_01;
+use FFI::CheckLib;
+
+requires qw<
+    ffi_subs_data
+    ffi_lib
+>;
+
+has 'ffi' => (
+    'is'       => 'ro',
+    'reader'   => '_ffi',
+    'init_arg' => undef,
+    'builder'  => '_build_ffi',
+);
+
+has 'ffi_subs_refs' => (
+    'is'       => 'ro',
+    'reader'   => '_ffi_subs_refs',
+    'init_arg' => undef,
+    'builder'  => '_build_ffi_subs_refs',
+    'lazy'     => 1,
+    'traits'   => ['Hash'],
+    'handles'  => { 'ffi_sub' => 'get' },
+);
+
+sub _build_ffi {
+    my $self     = shift;
+    my $ffi      = FFI::Platypus->new;
+    my $fallback = $self->can('ffi_alien');
+
+    $ffi->lib(
+        find_lib_or_exit(
+            'lib' => $self->ffi_lib,
```

Comment:
Pass `lib` in as a array ref like so:

```perl
  lib => [ $self->ffi_lib ],
```

internally it gets converted to an array ref if you provide a scalar anyway, and this allows ffi_lib to specify multiple libs as well.

Reply:
I guess ffi_lib could also return an array ref :man_shrugging: